src/presenter/new-event-presenter.js

```diff
@@ -0,0 +1,78 @@
+import {remove, render, RenderPosition} from '../framework/render.js';
+import {nanoid} from 'nanoid';
+import {UserAction, UpdateType, EVENT_TYPES} from '../const.js';
+import EditEventView from '../view/edit-event-view.js';
+
+export default class NewEventPresenter {
+  #eventListElement = null;
+  #handleDataChange = null;
+  #handleDestroy = null;
+  #destinations = null;
+  #offers = null;
+  #eventTypes = EVENT_TYPES;
+  #event = {};
+
+  #addEventElement = null;
+
+  constructor({eventListElement, destinations, offers, onDataChange, onDestroy}) {
+    this.#eventListElement = eventListElement;
+    this.#handleDataChange = onDataChange;
+    this.#handleDestroy = onDestroy;
+    this.#destinations = destinations;
+    this.#offers = offers;
+    this.#event = {...this.#event, type: 'flight', basePrice: '0', offers: []};
+  }
+
+  init() {
+    if (this.#addEventElement !== null) {
+      return;
+    }
+
+    this.#addEventElement = new EditEventView({
+      event: this.#event,
+      offers: this.#offers,
+      eventTypes: this.#eventTypes,
+      destinations: this.#destinations,
+      onFormSubmit: this.#handleFormSubmit,
+      onDeleteClick: this.#handleDeleteClick,
+      onFormClose: this.#handleDeleteClick,
+    });
+
+    render(this.#addEventElement, this.#eventListElement, RenderPosition.AFTERBEGIN);
+
+    document.addEventListener('keydown', this.#escKeyDownHandler);
+  }
+
+  destroy() {
+    if (this.#addEventElement === null) {
+      return;
+    }
+
+    this.#handleDestroy();
+
+    remove(this.#addEventElement);
+    this.#addEventElement = null;
+
+    document.removeEventListener('keydown', this.#escKeyDownHandler);
+  }
+
+  #handleFormSubmit = (event) => {
+    this.#handleDataChange(
+      UserAction.ADD_EVENT,
+      UpdateType.MINOR,
+      {id: nanoid(), ...event},
+    );
+    this.destroy();
+  };
+
+  #handleDeleteClick = () => {
+    this.destroy();
+  };
+
+  #escKeyDownHandler = (evt) => {
+    if (evt.key === 'Escape' || evt.key === 'Esc') {
```

Comment:
`Esc` - это keycode (старая API, которая уже depreacted), поэтому нет смысла делать вторую часть проверки, т.к. она никогда не будет `true`

```suggestion
    if (evt.key === 'Escape') {
```